File: public/snippets/C++/Switch Statements.cpp
#include <iostream>
using namespace std;

/* -------------------------------------------------- */
/*      SYNTAX SWITCH STATEMENT PADA BAHASA C++       */
/* -------------------------------------------------- */
/*  switch(expression) {                              */
/*                                                    */
/*     case constant-expression  :                    */
/*        statement(s);                               */
/*        break; // optional                          */ 
/*                                                    */
/*     case constant-expression  :                    */
/*        statement(s);                               */
/*        break; // optional                          */
/*                                                    */
/*     default : // Optional                          */
/*     statement(s);                                  */
/*  }                                                 */
/* -------------------------------------------------- */
 
int main () {
   // Deklarasi variabel lokal:
   char grade = 'D';

   switch(grade) {
      case 'A' :
         cout << "Excellent!" << endl; 
         break;
      case 'B' :
      case 'C' :
         cout << "Well done" << endl;
         break;
      case 'D' :
         cout << "You passed" << endl;
         break;
      case 'F' :
         cout << "Better try again" << endl;
         break;
      default :
        cout << "Invalid grade" << endl;
   }
	
   cout << "Your grade is " << grade << endl;
 
   return 0;
}

File: public/snippets/C#/Switch Statements.cs
using System;

/* -------------------------------------------------- */
/*      SYNTAX SWITCH STATEMENT PADA BAHASA C#        */
/* -------------------------------------------------- */
/*  switch(expression) {                              */
/*                                                    */
/*     case constant-expression  :                    */
/*        statement(s);                               */
/*        break; // optional                          */ 
/*                                                    */
/*     case constant-expression  :                    */
/*        statement(s);                               */
/*        break; // optional                          */
/*                                                    */
/*     default : // Optional                          */
/*     statement(s);                                  */
/*  }                                                 */
/* -------------------------------------------------- */

namespace DecisionMaking
{
   class Program
   {
      static void Main(string[] args)
      {
         /* Deklarasi variabel lokal */
         char grade = 'B';
         
         switch (grade)
         {
            case 'A':
               Console.WriteLine("Excellent!");
               break;
            case 'B':
            case 'C':
               Console.WriteLine("Well done");
               break;
            case 'D':
               Console.WriteLine("You passed");
               break;
            case 'F':
               Console.WriteLine("Better try again");
               break;
               default:
            Console.WriteLine("Invalid grade");
               break;
         }
         Console.WriteLine("Your grade is  {0}", grade);
         Console.ReadLine();
      }
   }
}